src/TLuaInterpreter.cpp

```diff
@@ -1309,6 +1309,21 @@ int TLuaInterpreter::centerview(lua_State* L)
     }
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getView
+int TLuaInterpreter::getView(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    if (!host.mpMap || !host.mpMap->mpRoomDB || !host.mpMap->mpMapper) {
+        lua_pushnil(L);
+        lua_pushstring(L, "you haven't opened a map yet");
+        return 2;
+    }
+
+    lua_pushnumber(L, host.mpMap->mRoomIdHash[host.getName()]);
```

Comment:
It still returns the last known room when the area is deleted.

That's a good suggestion, I'll change it.

Reply:
Not sure it is wise to use the `operator[]` on `host.mpMap->mRoomIdHash` remember it will create a default value member within that `QMap` should it not have a value - fair enough for there to ***be*** a map and the current profile not to have a player room set for it is a very small corner case - but if that is the case (for a *newly created* map for instance) lets not confuse the system by accidentally putting the player in a room with Id of `0` especially as that is NOT a valid room number...

I.e. use `host.mpMap->mRoomIdHash.value(host.getName(), -1)` not `host.mpMap->mRoomIdHash[host.getName()]` to look up the room the player is in for the current profile...

TBH It is probably sensible to check that the value return is a positive number and then to check that it exists and to return a nil + error message (`the player does not have a valid room id number set currently`) if it doesn't - note that the setter (currently only `centerview`) objects and fails if it is told to set the room id number to an invalid (non-existent) room!

Reply:
Applied both changes, thanks!